src/models/event.js

```
@@ -0,0 +1,193 @@
+eventSchema.index({ eventType: 1 });
+eventSchema.index({ fights: 1 });
+
+//< -------- VALIDATIONS ---------->
+eventSchema.path('ticketTypes').validate(function (value) {
+  return value.length > 0;
+}, 'Please provide at least one ticket type.');
+
+eventSchema.path('ticketTypes').validate(function (value) {
+  const ticketSet = new Set(value.map((ticket) => ticket.type));
+  return ticketSet === value;
+}, 'Each ticket type must be unique.');
```

Biome (1.9.4)

[error] This function expression can be turned into an arrow function.

Function expressions that don't use this can be turned into arrow functions.
Safe fix: Use an arrow function instead.


(lint/complexity/useArrowFunction)